postgresql/custom_types.go

```diff
@@ -66,202 +65,133 @@ func (j *JSONB) Scan(src interface{}) error {
 
 // Value implements the driver.Valuer interface.
 func (j JSONB) Value() (driver.Value, error) {
+	// See https://github.com/lib/pq/issues/528#issuecomment-257197239 on why are
+	// we returning string instead of []byte.
 	if j.V == nil {
 		return nil, nil
 	}
 	if v, ok := j.V.(json.RawMessage); ok {
 		return string(v), nil
 	}
-	return json.Marshal(j.V)
+	b, err := json.Marshal(j.V)
+	if err != nil {
+		return nil, err
+	}
+	return string(b), nil
 }
 
-// StringArray represents a PostgreSQL's varchar array.
-type StringArray []string
+// Type StringArray is an alias for pq.StringArray
+type StringArray pq.StringArray
+
+// Value implements the driver.Valuer interface.
+func (a StringArray) Value() (driver.Value, error) {
```

Comment:
Yes, the user can import `lib/pq` directly, these types are like proxies for pq, so the user can just import our `postgresql` package only.